oeq-ts-rest-api/src/AxiosInstance.ts

```diff
@@ -22,16 +22,26 @@ const catchHandler = (error: AxiosError | Error): never => {
  * @param queryParams The query parameters to send with the GET request
  * @param transformer A function which returns a copy of the raw data from the GET with any required values transformed - this should NOT mutate the input data (transforms should start on a copy/clone of the input)
  */
-export const GET = <T>(
+export function GET<T>(path: string): Promise<unknown>;
+export function GET<T>(
   path: string,
-  validator?: (data: unknown) => boolean,
+  validator: (data: unknown) => data is T,
   queryParams?: object,
-  transformer?: (data: unknown) => T,
-): Promise<T> =>
-  axios
-    .get(path, {params: queryParams, paramsSerializer: params => stringify(params)}, )
-    .then((response: AxiosResponse) => {
-      const data: any = transformer? transformer(response.data) : response.data;
+  transformer?: (data: unknown) => T
+): Promise<T>;
```

Comment:
this overloads the typings for the method `GET`

when no validator is present, it will return `Promise<undefined>`
when a validator is passed it will return `Promise<T>`

Reply:
I see. So based on what parameters are passed in we can expect what will be returned, either a Promise<T> or a Promise<unknown>.

Reply:
I did not know overload function can return different types. Thanks!